crates/primitives/src/chain/spec.rs

```diff
@@ -581,6 +584,35 @@ impl ForkCondition {
     }
 }
 
+/// Clap value parser for [ChainSpec]s that takes either a built-in chainspec or the path
+/// to a custom one.
+pub fn chain_spec_value_parser(s: &str) -> Result<ChainSpec, eyre::Error> {
```

Comment:
This is CLI specific so should not be here

Reply:
reverted, it was used in one test for merkletrie I fixed that too.